### PR TITLE
validator(fixed 2.0)

### PR DESCRIPTION
**Pull request by MaximShcherbachenko**

### CODE REVIEW

File: shcherbachenkoma/description.txt [hunk 1]
```diff
@@ -0,0 +1,6 @@
+name - Щербаченко Максим
```
Comment (igortereshchenko) [inline shcherbachenkoma/description.txt]: name - Щербаченко Максим 
group - KM - 93

видалити з файлу, тут тільки опис атрибутів

File: shcherbachenkoma/dataset_insert_manual.py [hunk 2]
```diff
@@ -0,0 +1,3 @@
+def add_info(patient, dataset):
+    dataset[patient] = {"agency_name": input("agency_name:"), "average_hcc_score": float(input("average_hcc_score:")),
+                        "percent_of_beneficiaries_with_asthma": int(input("percent_of_beneficiaries_with_asthma:"))}
```
Comment (GasperPaul) [inline shcherbachenkoma/dataset_insert_manual.py]: This should not use `input`

File: shcherbachenkoma/recursion.py [hunk 3]
```diff
@@ -0,0 +1,13 @@
+def recursion(dataset_copy):
+    if len(dataset_copy) == 0:
+        return None
+    else:
+        for provider_id in dataset_copy.keys():
```
Comment (GasperPaul) [inline shcherbachenkoma/recursion.py]: Why use `for`? Is this really a recursion then? Consiedr this:
```python
a = [1, 2, 3]
def rprint(lst):
    if len(lst) == 0: 
        return
    print(lst[0])
    rprint(lst[1:])
```
How can you write something like this?

File: shcherbachenkoma/recursion.py [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+            agancy_name = dataset_copy[provider_id]['agency_name']
+            average_hcc_score = float(dataset_copy[provider_id]['average_hcc_score'])
+            condition1 = float(dataset_copy[provider_id]['average_hcc_score']) > 2
+            condition2 = int(dataset_copy[provider_id]['percent_of_beneficiaries_with_asthma']) > 20
+            if condition1 or condition2:
+                del dataset_copy[provider_id]
```
Comment (GasperPaul) [inline shcherbachenkoma/recursion.py]: You should not modify the dataset.

File: shcherbachenkoma/validator/lib.py [hunk 5]
```diff
@@ -0,0 +1,24 @@
+import re
+re_provider_id = re.compile("^[+]?\d{6}?")
+re_agency_name = re.compile("^(([a-wA-W]+)\s{0,1})+")
+re_average_hcc_scor = re.compile("^([-+]?\d+\.?\d*\s*)+$")
+re_percent_of_beneficiaries_with_asthma = re.compile("^\d+$")
+def provider_id_validator(regexp):
+    value = input("Enter provider id:\n")
```
Comment (GasperPaul) [inline shcherbachenkoma/validator/lib.py]: I would expect you not to use `input` but just provide value as an argument.

File: shcherbachenkoma/validator/lib.py [hunk 6]
```diff
@@ -0,0 +1,24 @@
+import re
+re_provider_id = re.compile("^[+]?\d{6}?")
+re_agency_name = re.compile("^(([a-wA-W]+)\s{0,1})+")
+re_average_hcc_scor = re.compile("^([-+]?\d+\.?\d*\s*)+$")
+re_percent_of_beneficiaries_with_asthma = re.compile("^\d+$")
+def provider_id_validator(regexp):
```
Comment (GasperPaul) [inline shcherbachenkoma/validator/lib.py]: Does this regexp really should be an arg?

File: shcherbachenkoma/validator/lib.py [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+    value = input("Enter provider id:\n")
+    while not regexp.match(value) or len(value) > 6:
+        value = input("Must consist only of numbers (exactly six).\nTry again:\n")
+
```
Comment (GasperPaul) [inline shcherbachenkoma/validator/lib.py]: Why are you not returning anything?

File: shcherbachenkoma/dataset_insert_manual.py [hunk 8]
```diff
@@ -1,3 +1,3 @@
 def add_info(patient, dataset):
-    dataset[patient] = {"agency_name": input("agency_name:"), "average_hcc_score": float(input("average_hcc_score:")),
-                        "percent_of_beneficiaries_with_asthma": int(input("percent_of_beneficiaries_with_asthma:"))}
+    dataset[patient] = {"agency_name": "UNITED AMERICA HOME HEALTH SERVICES, INC.", "average_hcc_score": 2.24,
+                        "percent_of_beneficiaries_with_asthma": 19}
```
Comment (GasperPaul) [inline shcherbachenkoma/dataset_insert_manual.py]: Wouldn't it be great to have this values as params?

File: shcherbachenkoma/validator/lib.py [hunk 9]
```diff
@@ -3,22 +3,20 @@
 re_agency_name = re.compile("^(([a-wA-W]+)\s{0,1})+")
 re_average_hcc_scor = re.compile("^([-+]?\d+\.?\d*\s*)+$")
 re_percent_of_beneficiaries_with_asthma = re.compile("^\d+$")
-def provider_id_validator(regexp):
-    value = input("Enter provider id:\n")
-    while not regexp.match(value) or len(value) > 6:
-        value = input("Must consist only of numbers (exactly six).\nTry again:\n")
+def provider_id_validator(value):
+    if not re_provider_id.match(value) or len(value) > 6:
+        return False
```
Comment (GasperPaul) [inline shcherbachenkoma/validator/lib.py]: I think, this is missing something. By default, function returns `None`.

File: shcherbachenkoma/validator/test.py [hunk 10]
```diff
@@ -1,7 +1,7 @@
...
 percent_of_beneficiaries_with_asthma_validator
 
-provider_id_validator(re_provider_id)
-agency_name_validator(re_agency_name)
-average_hcc_score_validator(re_average_hcc_scor)
-percent_of_beneficiaries_with_asthma_validator(re_percent_of_beneficiaries_with_asthma)
+provider_id_validator('111111')
+agency_name_validator("UNITED AMERICA HOME HEALTH SERVICES, INC.")
+average_hcc_score_validator('2.24')
+percent_of_beneficiaries_with_asthma_validator('19')
```
Comment (GasperPaul) [inline shcherbachenkoma/validator/test.py]: How exactly should I use the return value of these validators? It's either `False` or `None`.

File: shcherbachenkoma/validator/lib.py (hunk 7 above)
Comment (GasperPaul) [reply shcherbachenkoma/validator/lib.py]: This still holds.

File: shcherbachenkoma/validator/lib.py (hunk 9 above)
Comment (GasperPaul) [reply shcherbachenkoma/validator/lib.py]: fixed

File: shcherbachenkoma/validator/lib.py (hunk 7 above)
Comment (GasperPaul) [reply shcherbachenkoma/validator/lib.py]: fixed

**Pull request merged**